File: Python3/1 - 200/66 Plus One.py
# 52ms, 91.53%
class Solution:
    def plusOne(self, digits):
        i, is_carry = -1, False
        while i > -len(digits) - 1:
            if (i == -1 or is_carry) and digits[i] == 9:
                digits[i] = 0
                is_carry = True
                i -= 1
            else:
                digits[i] += 1
                return digits
        digits.insert(0, 1)
        return digits
        

File: 1 - 200/66 Plus One.cc
// 4ms, 7.49%
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<int> plusOne(vector<int> &digits) {
        vector<int> ans(digits);
        for (int i = ans.size() - 1; i >= 0; --i) {
            ans[i] = (ans[i] + 1) % 10;
            if (ans[i] != 0) break;
            if (i == 0) ans.insert(ans.cbegin(), 1);
        }
        return ans;
    }
};
